modules/andes-core/broker/src/main/java/org/wso2/andes/store/file/FileMessageStoreImpl.java

```
@@ -0,0 +1,1364 @@
+
+import static org.wso2.andes.store.file.FileStoreConstants.DLC;
+import static org.wso2.andes.store.file.FileStoreConstants.DLC_QUEUE_ID;
+
+/**
+ * Implementation of LevelDB broker store. Message persistence related methods are implemented
+ * in this class.
+ */
+public class FileMessageStoreImpl implements MessageStore {
+
+    private static final Logger log = Logger.getLogger(FileMessageStoreImpl.class);
+
+    /**
+     * All the database operations are preformed on this object. Refers to the LevelDB store.
+     */
+    private DB brokerStore;
+
+    /**
+     * Used to handle the last queue id.
+     */
+    private int lastQueueID;
+
+    /**
+     * {@inheritDoc} Check if data can be inserted, read and finally deleted
+     * from the database.
+     */
+    @Override
+    public boolean isOperational(String testString, long testTime) {
+        return false;
+    }
+
+    /**
+     * {@inheritDoc} block restart interval, block size, cache size, max open files, write buffer size and path of the
+     * leveldb store can be configured.
+     */
+    @Override
+    public DurableStoreConnection initializeMessageStore(AndesContextStore contextStore, ConfigurationProperties
+            connectionProperties) throws AndesException {
+        FileStoreConnection fileStoreConnection = new FileStoreConnection();
+        fileStoreConnection.initialize(connectionProperties);
+        this.brokerStore = (DB) fileStoreConnection.getConnection();
+        return fileStoreConnection;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void storeMessages(List<AndesMessage> messageList) throws AndesException {
+        Transaction tx = new Transaction(brokerStore);
+
+        try {
+            for (AndesMessage message : messageList) {
+                storeMessage(message, tx);
+            }
+            tx.commit(brokerStore);
+        } catch (DBException e) {
+            log.error("Messages storing failed", e);
```

Shouldn't tx.rollback here?

---

We should also throw an exception here.